src/views/Pool.vue

```diff
@@ -106,6 +88,44 @@ export default {
     };
   },
   computed: {
+    collateralInfo() {
+      return this.$store.getters.getCollateralInfo;
+    },
+    userBalancesProp() {
+      const pool = this.pool;
+      const balances = [
+        {
+          balance: this.userBalanceNativeToken,
+          token: "AVAX",
+          decimals: "18",
+        },
+        {
+          balance: this.userBalanceToken,
+          token: pool.token.name,
+          decimals: pool.token.decimals.toString(),
+        },
+        {
+          balance: this.userBalancePairToken,
+          token: pool.pairToken.name,
+          decimals: pool.pairToken.decimals.toString(),
+        },
+      ];
+      if (pool.userBalanceNativeToken === "") {
+        const filterBalances = balances.filter((item) => item !== "");
+        return filterBalances;
```

Comment:
All items corresponds to the condition, so it won't filtered anything.

Reply:
In case of `pool.userBalanceNativeToken === ""` (by the way, it is better to compare it with `undefined` instead of an empty string), you can just return slice, no need to filter it.

Reply:
Сhanged the filter to a slice, put the logic with the slice in the top component